packages/bson-transpilers/symbols/go/templates.yaml

```diff
@@ -0,0 +1,912 @@
+                return `"${str.replace(/\\([\s\S])|(")/g, '\\$1$2')}"`;
+            }
+
+            const structParts = [];
+            structParts.push(`Pattern: ${doubleStringify(pattern)}`);
+            if (flags)
+                structParts.push(`Options: ${doubleStringify(flags)}`);
+            return `(${structParts.join(", ")})`;
+        }
+    Decimal128SymbolTemplate: &Decimal128SymbolTemplate !!js/function >
+        () => ''
+    Decimal128SymbolArgsTemplate: &Decimal128SymbolArgsTemplate !!js/function >
+        (_, arg) => {
+            if (!arg)
+                arg = '"0"';
+
+            // Double quote stringify
+            const singleQuoted = arg.charAt(0) === '\'' && arg.charAt(arg.length - 1) === '\'';
+            const doubleQuoted = arg.charAt(0) === '"' && arg.charAt(arg.length - 1) === '"';
+            if (singleQuoted || doubleQuoted)
+                arg = arg.substr(1, arg.length - 2);
+            arg = `"${arg.replace(/\\([\s\S])|(")/g, '\\$1$2')}"`;
+
+            this.declarations.addFunc([]
+                .concat('var parseDecimal128 = func(str string) primitive.Decimal128 {')
+                .concat('  d128, err := primitive.ParseDecimal128(str)')
+                .concat('  if err != nil {')
+                .concat('    log.Fatal(err)')
+                .concat('  }')
+                .concat('  return d128')
+                .concat('}')
+                .join('\n'));
+            return `parseDecimal128(${arg})`;
+        }
+    MinKeySymbolTemplate: &MinKeySymbolTemplate !!js/function >
+        () => 'primitive.MinKey'
+    MinKeySymbolArgsTemplate: &MinKeySymbolArgsTemplate !!js/function >
+        () => '{}'
+    MaxKeySymbolTemplate: &MaxKeySymbolTemplate !!js/function >
+        () => 'primitive.MaxKey'
+    MaxKeySymbolArgsTemplate: &MaxKeySymbolArgsTemplate !!js/function >
+        () => '{}'
+    TimestampSymbolTemplate: &TimestampSymbolTemplate !!js/function >
+        () => 'primitive.Timestamp'
+    TimestampSymbolArgsTemplate: &TimestampSymbolArgsTemplate !!js/function >
+        (lhs, low, high) => {
+            if (low === undefined) {
+                low = 0;
+                high = 0;
+            }
+            return `{T: ${low}, I: ${high}}`
+        }
+    # non bson-specific
+    NumberSymbolTemplate: &NumberSymbolTemplate null
+    NumberSymbolArgsTemplate: &NumberSymbolArgsTemplate null
+    DateSymbolTemplate: &DateSymbolTemplate !!js/function >
+        () => 'time.Date'
+    DateSymbolArgsTemplate: &DateSymbolArgsTemplate !!js/function >
+        (lhs, date, isString) => {
+            if (date === null)
+                return `${lhs}.Now()`;
+
+            const dateStr = [
+                date.getUTCFullYear(),
+                date.getUTCMonth() + 1,
+                date.getUTCDate(),
+                date.getUTCHours(),
+                date.getUTCMinutes(),
+                date.getUTCSeconds(),
+                'nil',
```

Comment:
The `nsec` value must be an int, so I believe this should be `'0'` instead of `'nil'`.

E.g. filter:
```
{x: ISODate('2020-01-01')}
```
and generated Go code:
```go
bson.D{{"x", time.Date(2020, 1, 1, 0, 0, 0, nil, time.UTC)}}
```
Compilation error:
```
cannot use nil as int value in argument to time.Date
```